code/modules/ai/ai_minds/carbon/xeno/xeno.dm

```diff
@@ -0,0 +1,112 @@
+				break
+
+		//Cheat mode: insta open airlocks
+		for(var/obj/machinery/door/airlock/lock in range(mob_parent, 1))
+			if(lock && lock.density && !lock.operating && !lock.welded)
+				lock.open()
+				return //Don't try going on window frames dammit
+
+		for(var/obj/structure/window in range(mob_parent, 1))
+			if(istype(window, /obj/structure/window_frame)) //Teleport onto that window
+				mob_parent.loc = window.loc
+				break
+
+/datum/ai_mind/carbon/xeno/get_new_state(reason_for)
+	if(reason_for == REASON_TARGET_KILLED || reason_for == REASON_FINISHED_NODE_MOVE || reason_for == REASON_TARGET_SPOTTED || reason_for == REASON_REFRESH_TARGET)
+		var/list/potential_targets = get_targets() //Define here as if there's targets we don't need to redundently call the get targets
+		if(length(potential_targets)) //There's targets nearby, kill the closest thing
+			var/closest_dist = 999
+			var/atom/favorable_target
+			for(var/atom/a in potential_targets)
+				if(get_dist(mob_parent, a) <= closest_dist)
+					closest_dist = get_dist(mob_parent, a)
+					favorable_target = a
+			if(favorable_target)
+				atom_to_walk_to = favorable_target
+				cur_action_state = /datum/element/action_state/move_to_atom
+				return list(cur_action_state, atom_to_walk_to, distance_to_maintain, sidestep_prob)
+			else
+				reason_for = REASON_TARGET_KILLED //Probably no more targets
+		else //No targets, let's just randomly move to nodes
+			return ..()
+	else //If we get here reason_for is null, let parent handle that
+		return ..()
+
+/datum/ai_mind/carbon/xeno/get_signals_to_reg()
+	if(istype(atom_to_walk_to, /mob/living/carbon/human))
+		return list(
+				list(mob_parent, COMSIG_CLOSE_TO_MOB, /datum/component/ai_behavior/.proc/mind_attack_target),
+				list(atom_to_walk_to, COMSIG_MOB_DEATH, /datum/component/ai_behavior/.proc/reason_target_killed)
+				)
+
+	if(istype(atom_to_walk_to, /obj/machinery))
+		return list(
+				list(mob_parent, COMSIG_CLOSE_TO_MACHINERY, /datum/component/ai_behavior/.proc/mind_attack_target),
+				list(atom_to_walk_to, COMSIG_PARENT_QDELETING, /datum/component/ai_behavior/.proc/reason_target_killed)
+				)
+
+	return ..() //Walking to a node
+
+/datum/ai_mind/carbon/xeno/get_signals_to_unreg()
+	if(atom_to_walk_to)
+
+		if(istype(atom_to_walk_to, /mob/living/carbon/human))
+			return list(
+					list(mob_parent, COMSIG_CLOSE_TO_MOB),
+					list(atom_to_walk_to, COMSIG_MOB_DEATH)
+					)
+
+		if(istype(atom_to_walk_to, /obj/machinery))
+			return list(
+					list(mob_parent, COMSIG_CLOSE_TO_MACHINERY),
+					list(atom_to_walk_to, COMSIG_PARENT_QDELETING)
+					)
+	return ..() //Walking to a node
+
+/datum/ai_mind/carbon/xeno/attack_target()
+	if(get_dist(atom_to_walk_to, mob_parent) <= attack_range && world.time > mob_parent.next_move)
+		var/mob/living/carbon/xenomorph/xeno = mob_parent
+		if(istype(atom_to_walk_to, /mob))
+			var/mob/target = atom_to_walk_to //Typed for parameter
+			target.attack_alien(xeno, force_intent = INTENT_HARM)
```

Comment:
you'll need to change /atom/proc/attack_alien() to have the full set of parameters as this
`/mob/living/attack_alien(mob/living/carbon/xenomorph/X, dam_bonus, set_location = FALSE, random_location = FALSE, no_head = FALSE, no_crit = FALSE, force_intent = null)`

Reply:
Had the ai mind do a intent set rather than forcing it through parameters